File: src/main/kotlin-graphql/io/github/graphql/kotlin/model/TagTO.kt
package io.github.graphql.kotlin.model

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLRequestSerializer
import java.util.Objects
import java.util.StringJoiner

@javax.annotation.Generated(
    value = ["com.kobylynskyi.graphql.codegen.GraphQLCodegen"],
    date = "2021-03-07T19:48:13+0800"
)
open class TagTO(
    override
    val abbreviatedOid: String,
    override
    val commitResourcePath: String,
    override
    val commitUrl: String,
    override
    val id: String,
    val message: String?,
    val name: String,
    override
    val oid: String,
    override
    val repository: RepositoryTO,
    val tagger: GitActorTO?,
    val target: GitObjectTO
) : GitObjectTO, NodeTO {

    companion object {
        fun builder(): Builder = Builder()
    }

    // In the future, it maybe change.
    override fun toString(): String {
        val joiner = StringJoiner(", ", "{ ", " }")
        joiner.add("abbreviatedOid: " + GraphQLRequestSerializer.getEntry(abbreviatedOid))
        joiner.add("commitResourcePath: " + GraphQLRequestSerializer.getEntry(commitResourcePath))
        joiner.add("commitUrl: " + GraphQLRequestSerializer.getEntry(commitUrl))
        joiner.add("id: " + GraphQLRequestSerializer.getEntry(id))
        if (message != null) {
            joiner.add("message: " + GraphQLRequestSerializer.getEntry(message))
        }
        joiner.add("name: " + GraphQLRequestSerializer.getEntry(name))
        joiner.add("oid: " + GraphQLRequestSerializer.getEntry(oid))
        joiner.add("repository: " + GraphQLRequestSerializer.getEntry(repository))
        if (tagger != null) {
            joiner.add("tagger: " + GraphQLRequestSerializer.getEntry(tagger))
        }
        joiner.add("target: " + GraphQLRequestSerializer.getEntry(target))
        return joiner.toString()
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }
        if (other == null || javaClass != other.javaClass) {
            return false
        }
        val that = other as TagTO
        return Objects.equals(abbreviatedOid, that.abbreviatedOid)
                && Objects.equals(commitResourcePath, that.commitResourcePath)
                && Objects.equals(commitUrl, that.commitUrl)
                && Objects.equals(id, that.id)
                && Objects.equals(message, that.message)
                && Objects.equals(name, that.name)
                && Objects.equals(oid, that.oid)
                && Objects.equals(repository, that.repository)
                && Objects.equals(tagger, that.tagger)
                && Objects.equals(target, that.target)

    }

    override fun hashCode(): Int = {
        return Objects.hash(abbreviatedOid, commitResourcePath, commitUrl, id, message, name, oid, repository, tagger, target)
    }

    class Builder {

        private lateinit var abbreviatedOid: String
        private lateinit var commitResourcePath: String
        private lateinit var commitUrl: String
        private lateinit var id: String
        private var message: String? = null
        private lateinit var name: String
        private lateinit var oid: String
        private lateinit var repository: RepositoryTO
        private var tagger: GitActorTO? = null
        private lateinit var target: GitObjectTO

        fun setAbbreviatedOid(abbreviatedOid: String): Builder {
            this.abbreviatedOid = abbreviatedOid
            return this
        }

        fun setCommitResourcePath(commitResourcePath: String): Builder {
            this.commitResourcePath = commitResourcePath
            return this
        }

        fun setCommitUrl(commitUrl: String): Builder {
            this.commitUrl = commitUrl
            return this
        }

        fun setId(id: String): Builder {
            this.id = id
            return this
        }

        fun setMessage(message: String?): Builder {
            this.message = message
            return this
        }

        fun setName(name: String): Builder {
            this.name = name
            return this
        }

        fun setOid(oid: String): Builder {
            this.oid = oid
            return this
        }

        fun setRepository(repository: RepositoryTO): Builder {
            this.repository = repository
            return this
        }

        fun setTagger(tagger: GitActorTO?): Builder {
            this.tagger = tagger
            return this
        }

        fun setTarget(target: GitObjectTO): Builder {
            this.target = target
            return this
        }

        fun build(): TagTO {
            return TagTO(abbreviatedOid, commitResourcePath, commitUrl, id, message, name, oid, repository, tagger, target)
        }
    }
}

File: src/main/scala-graphql/io/github/graphql/scala/model/TagTO.scala
package io.github.graphql.scala.model

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLRequestSerializer
import java.util.Objects
import scala.collection.JavaConverters._

@javax.annotation.Generated(
    value = Array("com.kobylynskyi.graphql.codegen.GraphQLCodegen"),
    date = "2021-03-07T19:49:36+0800"
)
class TagTO(
    @javax.validation.constraints.NotNull
    override val abbreviatedOid: String,
    @javax.validation.constraints.NotNull
    override val commitResourcePath: String,
    @javax.validation.constraints.NotNull
    override val commitUrl: String,
    @javax.validation.constraints.NotNull
    override val id: String,
    message: String,
    @javax.validation.constraints.NotNull
    name: String,
    @javax.validation.constraints.NotNull
    override val oid: String,
    @javax.validation.constraints.NotNull
    override val repository: RepositoryTO,
    tagger: GitActorTO,
    @javax.validation.constraints.NotNull
    target: GitObjectTO
) extends GitObjectTO with NodeTO {

    override def toString(): String = {
        Seq(
            if (abbreviatedOid != null) "abbreviatedOid: " + GraphQLRequestSerializer.getEntry(abbreviatedOid) else "",
            if (commitResourcePath != null) "commitResourcePath: " + GraphQLRequestSerializer.getEntry(commitResourcePath) else "",
            if (commitUrl != null) "commitUrl: " + GraphQLRequestSerializer.getEntry(commitUrl) else "",
            if (id != null) "id: " + GraphQLRequestSerializer.getEntry(id) else "",
            if (message != null) "message: " + GraphQLRequestSerializer.getEntry(message) else "",
            if (name != null) "name: " + GraphQLRequestSerializer.getEntry(name) else "",
            if (oid != null) "oid: " + GraphQLRequestSerializer.getEntry(oid) else "",
            if (repository != null) "repository: " + GraphQLRequestSerializer.getEntry(repository) else "",
            if (tagger != null) "tagger: " + GraphQLRequestSerializer.getEntry(tagger) else "",
            if (target != null) "target: " + GraphQLRequestSerializer.getEntry(target) else ""
        ).filter(_ != "").mkString("{", ",", "}")
    }

    override def equals(obj: Any): Boolean = {
        if (this == obj) {
            return true
        }
        if (obj == null || getClass != obj.getClass) {
            return false
        }
        val that = obj.asInstanceOf[TagTO]
        Objects.equals(abbreviatedOid, that.abbreviatedOid) &&
        Objects.equals(commitResourcePath, that.commitResourcePath) &&
        Objects.equals(commitUrl, that.commitUrl) &&
        Objects.equals(id, that.id) &&
        Objects.equals(message, that.message) &&
        Objects.equals(name, that.name) &&
        Objects.equals(oid, that.oid) &&
        Objects.equals(repository, that.repository) &&
        Objects.equals(tagger, that.tagger) &&
        Objects.equals(target, that.target)
    }

    override def hashCode(): Int = {
        Objects.hash(abbreviatedOid, commitResourcePath, commitUrl, id, message, name, oid, repository, tagger, target)
    }
}

object TagTO {

    def builder(): TagTO.Builder = new Builder()

    class Builder {

        private var abbreviatedOid: String = _
        private var commitResourcePath: String = _
        private var commitUrl: String = _
        private var id: String = _
        private var message: String = _
        private var name: String = _
        private var oid: String = _
        private var repository: RepositoryTO = _
        private var tagger: GitActorTO = _
        private var target: GitObjectTO = _

        def setAbbreviatedOid(abbreviatedOid: String): Builder = {
            this.abbreviatedOid = abbreviatedOid
            this
        }

        def setCommitResourcePath(commitResourcePath: String): Builder = {
            this.commitResourcePath = commitResourcePath
            this
        }

        def setCommitUrl(commitUrl: String): Builder = {
            this.commitUrl = commitUrl
            this
        }

        def setId(id: String): Builder = {
            this.id = id
            this
        }

        def setMessage(message: String): Builder = {
            this.message = message
            this
        }

        def setName(name: String): Builder = {
            this.name = name
            this
        }

        def setOid(oid: String): Builder = {
            this.oid = oid
            this
        }

        def setRepository(repository: RepositoryTO): Builder = {
            this.repository = repository
            this
        }

        def setTagger(tagger: GitActorTO): Builder = {
            this.tagger = tagger
            this
        }

        def setTarget(target: GitObjectTO): Builder = {
            this.target = target
            this
        }

        def build(): TagTO = new TagTO(abbreviatedOid, commitResourcePath, commitUrl, id, message, name, oid, repository, tagger, target)

    }
}
